### ISSUE TITLE
🐛 Collaborator section has multiple bugs

### ISSUE THREAD

**Issue by kcullion** [icgc-argo/dac-ui#362]
## Describe the bugs

There are some weird things happening in the collaborators section/ethics sections. Logging them all here for investigation 

- [x] When filling out a collaborator and adding them to the table, the table of contents should be checked off - but you have to move away from it in order for that section to be checked off
![image](https://user-images.githubusercontent.com/18424138/129085144-c223cd32-f463-4f4b-b94b-d7391ab556da.png)

- [x] Also when a user doesn't add collaborators to the form, but fills out the rest of the form correctly, the collaborators used to automatically get checked off when the Sign and Submit section enables. That's not the case anymore - the user has to refresh the application to see that section checked off.
![image](https://user-images.githubusercontent.com/18424138/129084914-d935668c-d110-4d04-aa46-750c6461532d.png)

- [x] this modal may be related, but made a ticket: https://github.com/icgc-argo/dac-ui/issues/364
- [x] Sam found that clicking on the collaborator section leads to this error
![image](https://user-images.githubusercontent.com/18424138/129100678-0017cf2f-635c-4cc1-8579-d8f402b075ec.png)

- [x] There's also this error when clicking on ethics section that Sam found: 
![image](https://user-images.githubusercontent.com/18424138/129100943-1884e36f-92c2-4f6d-81d6-6e791c01d963.png)

- [x] When adding a collaborator, it adds to the table, but when I go to add a new collaborator, the modal data isn't cleared
![image](https://user-images.githubusercontent.com/18424138/129100843-e36d1962-23b8-4c14-aa8f-e215ac481b5b.png)

**Comment by caravinci** [comment]
This is happening because of the descentralised call to a different endpoint when we create a new collaborator, which is not done inside the validation routine.

What needs to happen is fairly simple: 
- this https://github.com/icgc-argo/dac-ui/blob/develop/components/pages/Applications/ApplicationForm/Forms/index.tsx#L89 
 - needs to be called from here https://github.com/icgc-argo/dac-ui/blob/develop/components/pages/Applications/ApplicationForm/Forms/Collaborators/index.tsx#L128
 
 It's only a matter of passing that call down to Collaborators.

**Comment by caravinci** [comment]
Coincidentally, I've found the `value.map` while working on #361 and fixed it there.

**Comment by caravinci** [comment]
The issues with the modal form not getting cleared and the delete modal not showing the collaborator's name are resolved via #329

**Comment by caravinci** [comment]
Could not replicate the first error Sam found (`Cannot convert undefined or null to object`) at all.

**Comment by caravinci** [comment]
Fixes implemented rely on #365 being merged first.

**Comment by kcullion** [comment]
Tested Aug 12/13 
- Table of contents icon behaves as expected
- Don't see the undefined values anymore
- Modal is cleared after adding a new collaborator
- Errors that Sam found are resolved.

we just found some log errors in the console so I'm making a new ticket for @samrichca to resolve that. https://github.com/icgc-argo/dac-ui/issues/373

Regarding: `As a sidenote, I did notice if you save the email as the collaborator's first, and then go an add the email as the applicant's, API is not tagging the collaborator as invalid/errored the same way it does for Affiliation. Not sure there's a ticket for that already`
We made a ticket for Ann during sprint planning for this: https://github.com/icgc-argo/dac-api/issues/151
